Misc docs cleanup

Zac-HD opened the pull request:
No associated issue, just a pile of small cleanups I've had accumulating for a while.

DRMacIver commented on hypothesis-python/RELEASE.rst:
```diff
@@ -0,0 +1,3 @@
+RELEASE_TYPE: patch
+
+This patch contains some internal refactoring.  There are no user-visible changes.
```
This isn't true. Documentation changes are user visible!

DRMacIver commented on hypothesis-python/src/hypothesis/searchstrategy/recursive.py:
```diff
@@ -101,9 +100,6 @@ def do_draw(self, data):
-                            "Draw for %r exceeded max_leaves and had to be retried",
-                            self,
-                        )
+                        "Draw for %r exceeded max_leaves and had to be retried" % self
```
Are you sure this is pointless? It might be, but I think it needs more of an argument that this is the case - the problem `lazyformat` solves is that the `repr` for strategies may be (very) expensive to evaluate, and we don't want to calculate it each time we call `note_event` unless we need to.

DRMacIver commented on hypothesis-python/docs/packaging.rst:
```diff
@@ -30,14 +30,10 @@ Dependencies
+Hypothesis is designed to work with a range of Python versions.  We always support
+`all versisions of CPython with upstream support <https://devguide.python.org/#status-of-python-branches>`_,
+and plan to drop Python 2 at EOL in 2020.  We also support the latest versions of PyPy
+for Python 2 and Python 3.
```
Inconsistent with what it says elsewhere (and our informal plans) that we're dropping Python 2 support in PyPy when we drop it for CPython even though PyPy will continue to support it.

Zac-HD commented:
OK!  I've just dropped the lazyformat removal for now, and will leave well enough alone unless I can someday be bothered running a profiler to see what it's doing.

So this is a docs-only patch with no release number, and I've fixed the wording for PyPy2.
